test/e2e/upgradetest/upgrade_test.go

```diff
@@ -312,11 +312,92 @@ func testBackupForOldClusterWithBackupPolicy(t *testing.T, bp *spec.BackupPolicy
 	}
 }
 
-// get the image of the first container of the pod in the deployment.
-func getContainerImageNameFromDeployment(name string) (string, error) {
-	d, err := testF.KubeCli.AppsV1beta1().Deployments(testF.KubeNS).Get(name, metav1.GetOptions{})
+// TestDisasterRecovery tests if the new operator could do disaster recovery from backup of the old cluster.
+func TestDisasterRecovery(t *testing.T) {
+	t.Run("Recovery from PV backup", func(t *testing.T) {
+		testDisasterRecoveryWithBackupPolicy(t, e2eutil.NewPVBackupPolicy())
+	})
+	t.Run("Recovery from S3 backup", func(t *testing.T) {
+		testDisasterRecoveryWithBackupPolicy(t, e2eutil.NewS3BackupPolicy())
+	})
+}
+
+func testDisasterRecoveryWithBackupPolicy(t *testing.T, bp *spec.BackupPolicy) {
+	// create operator
+	err := testF.CreateOperator()
+	if err != nil {
+		t.Fatalf("failed to create operator:%v", err)
+	}
+	defer func() {
+		err := testF.DeleteOperator()
+		if err != nil {
+			t.Fatalf("failed to delete operator:%v", err)
+		}
+	}()
+	err = k8sutil.WaitEtcdTPRReady(testF.KubeCli.CoreV1().RESTClient(), 3*time.Second, 30*time.Second, testF.KubeNS)
+	if err != nil {
+		t.Fatalf("failed to see cluster TPR get created in time: %v", err)
+	}
+
+	// create cluster
+	bp.BackupIntervalInSecond = 60 * 60 * 24 // long enough that no backup was made automatically
```

Comment:
Well we only really care about creating the backup after writing data to etcd. 
So we can't rely on a periodic backup by setting the `BackupIntervalInSecond ` since there's no guarantee if that will make a backup immediately after `PutDataToEtcd`.

That's why we avoid the periodic backup with a long timeout and just request the backup explicitly after `PutDataToEtcd`.